src/js/core.js

```diff
@@ -923,14 +923,38 @@ function MediumEditor(elements, options) {
                 i;
 
             if (opts.url && opts.url.trim().length > 0) {
-                this.options.ownerDocument.execCommand('createLink', false, opts.url);
-
-                if (this.options.targetBlank || opts.target === '_blank') {
-                    Util.setTargetBlank(Selection.getSelectionStart(this.options.ownerDocument), opts.url);
-                }
+                var currentSelection = this.options.contentWindow.getSelection();
+                if (currentSelection) {
+                    var exportedSelection = this.exportSelection(),
+                        currentEditor,
+                        startContainerParentElement,
+                        endContainerParentElement,
+                        textNodes;
+
+                    currentEditor = Util.traverseUp(currentSelection.getRangeAt(0).startContainer, function (node) {
+                        return node.getAttribute('data-medium-editor-element');
+                    });
+                    startContainerParentElement = Util.findParentBlockContainer(
+                        currentSelection.getRangeAt(0).startContainer);
+                    endContainerParentElement = Util.findParentBlockContainer(
+                        currentSelection.getRangeAt(0).endContainer);
+
+                    if (startContainerParentElement === endContainerParentElement) {
```

Comment:
Wasn't the goal of the code to target the case where the selection spans multiple block elements?  It seems like `startContianerParentElement === endContainerParentElement` would be for the most common case (where the selection is all within 1 paragraph) or have I misunderstood something?